### PR TITLE
DOP-1409: Support new AST structure for target IDs

### PR DESCRIPTION
**Pull request by sophstad**
Support upcoming breaking changes in snooty-parser (to be introduced in ~v.0.6.0 next week). Introduced in this PR: https://github.com/mongodb/snooty-parser/pull/204

Equivalent change in Snooty: https://github.com/mongodb/snooty/pull/271

### CODE REVIEW

**Comment by jestapinski** [comment]
[Here is a staging link for this work](https://docs-mongodbcom-staging.corp.mongodb.com/master/devhub/jordanstapinski/ss/DOP-1409/)! Going through it now

File: src/components/DefinitionListItem.js [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
-                <ComponentFactory
-                    {...rest}
-                    nodeData={child}
-                    key={`dd-${index}`}
-                />
-            ))}
-        </dd>
-    </React.Fragment>
-);
+const DefinitionListItem = ({ nodeData: { children, term }, ...rest }) => {
```
Comment (jestapinski) [inline src/components/DefinitionListItem.js:6]: Not sure if this one is being used in any articles? I looked for any `definitionList` directive but didn't see any in use, so this is likely fine :)

File: src/utils/find-key-value-pair.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Recursively searches child nodes to find the specified key/value pair.
+ * Prevents us from having to rely on a fixed depth for properties in the AST.
+ */
+const findKeyValuePair = (nodes, key, value) => {
+    let result;
+    const iter = node => {
+        if (node[key] === value) {
+            result = node;
```
Comment (jestapinski) [inline src/utils/find-key-value-pair.js:9]: Really like the scope-usage here!

**Review by jestapinski — APPROVED** [review]
Looks great from what I see! Thanks for this change, we will merge it in next week with the parser change

File: src/components/DefinitionListItem.js (hunk 1 above)
Comment (sophstad) [reply src/components/DefinitionListItem.js:6]: Gotcha! It actually mostly appears in `.. glossary::` directives which I also don't see in devhub, so yup should be all good!

**Comment by jestapinski** [comment]
Thanks!

**Pull request merged**